core/arch/arm/mm/core_mmu.c

```diff
@@ -1024,17 +1024,13 @@ TEE_Result core_mmu_map_pages(vaddr_t vstart, paddr_t *pages, size_t num_pages,
 		}
 
 		while (true) {
-			if (!core_mmu_find_table(vaddr, UINT_MAX, &tbl_info)) {
-				ret = TEE_ERROR_GENERIC;
-				goto err;
-			}
+			if (!core_mmu_find_table(vaddr, UINT_MAX, &tbl_info))
+				panic("Can't find pagetable for vaddr ");
 
 			idx = core_mmu_va2idx(&tbl_info, vaddr);
 			if (tbl_info.shift != SMALL_PAGE_SHIFT) {
 				if (!core_mmu_divide_block(&tbl_info, idx)) {
-					EMSG("Could not divide block into smaller tables");
-					ret = TEE_ERROR_OUT_OF_MEMORY;
-					goto err;
+					panic("Could not divide block into smaller tables");
 				}
 			} else {
 				break;
```

Comment:
a bit overcomplicated. what about (sorry, nitpicking on minor stylish stuff, but below is more straightforward):
```c
	while (true) {
		(...)

		if (tbl_info.shift == SMALL_PAGE_SHIFT)
			break;
		if (!core_mmu_divide_block(&tbl_info, idx)) {
			panic("Could not divide block into smaller tables");
	}